components/homme/cmake/Kokkos.cmake

```diff
@@ -1,12 +1,15 @@
 set (KOKKOS_INSTALLATION_NEEDED FALSE CACHE LOGICAL "")
-if (NOT DEFINED KOKKOS_PATH)
+if (NOT DEFINED E3SM_KOKKOS_PATH)
   # Build kokkos submodule if user did not specify KOKKOS_PATH.
   set (KOKKOS_INSTALLATION_NEEDED TRUE)
   set (KOKKOS_SRC ${CMAKE_SOURCE_DIR}/../../externals/kokkos)
   set (KOKKOS_INSTALL_DIR ${CMAKE_BINARY_DIR}/kokkos/install)
   SET (KOKKOS_INCLUDE_DIR ${KOKKOS_SRC}/core/src ${KOKKOS_INSTALL_DIR})
   SET (KOKKOS_LIBRARY_DIR ${KOKKOS_INSTALL_DIR})
+  message ("Kokkos will be built internally to E3SM, in ${KOKKOS_INSTALL_DIR}")
```

Comment:
"in ${KOKKOS_INSTALL_DIR}" is not quite true. We can't actually get Kokkos to install via the current monolithic add_subdirectory(kokkos) approach. I think it's better not to confuse the user with unnecessary and unusable details, anyway. I recommend you write just the first part: "Kokkos will be built internally to E3SM".

Reply:
The issue here is the name `KOKKOS_INSTALL_DIR`. Kokkos will actually be built there, but not installed. The printout message is correct though, namely it prints the actual folder where kokkos gets built.

We could change the names given to the cmake variables, namely `KOKKOS_INSTALL_DIR` renamed to `KOKKOS_BUILD_DIR`. And maybe to the folders too (e.g., `${CMAKE_BINARY_DIR}/kokkos/install` should probably be called `${CMAKE_BINARY_DIR}/kokkos/build`). Would you be fine with something like that?

Reply:
OK. Fine as is, or feel free to change. At this point, the PR mechanically works, so it's just a matter of taste.